pkg/operator/starter.go

```diff
@@ -0,0 +1,107 @@
+package operator
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/openshift/aws-efs-csi-driver-operator/assets"
+	"github.com/openshift/library-go/pkg/operator/loglevel"
+	"github.com/openshift/library-go/pkg/operator/managementstatecontroller"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+	kubeclient "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+
+	opv1 "github.com/openshift/api/operator/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	configinformers "github.com/openshift/client-go/config/informers/externalversions"
+	"github.com/openshift/library-go/pkg/controller/controllercmd"
+	"github.com/openshift/library-go/pkg/controller/manager"
+	"github.com/openshift/library-go/pkg/operator/csi/csidrivernodeservicecontroller"
+	goc "github.com/openshift/library-go/pkg/operator/genericoperatorclient"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+const (
+	// Operand and operator run in the same namespace
+	operatorName = "aws-efs-csi-driver-operator"
+	instanceName = "efs.csi.aws.com"
+
+	namespaceReplaceKey = "${NAMESPACE}"
+)
+
+func RunOperator(ctx context.Context, controllerConfig *controllercmd.ControllerContext) error {
+	operatorNamespace := controllerConfig.OperatorNamespace
+
+	// Create core clientset and informer
+	kubeClient := kubeclient.NewForConfigOrDie(rest.AddUserAgent(controllerConfig.KubeConfig, operatorName))
+	kubeInformersForNamespaces := v1helpers.NewKubeInformersForNamespaces(kubeClient, operatorNamespace, "")
+
+	// Create config clientset and informer. This is used to get the cluster ID
+	configClient := configclient.NewForConfigOrDie(rest.AddUserAgent(controllerConfig.KubeConfig, operatorName))
+	configInformers := configinformers.NewSharedInformerFactory(configClient, 20*time.Minute)
+
+	// Create GenericOperatorclient. This is used by the library-go controllers created down below
+	gvr := opv1.SchemeGroupVersion.WithResource("clustercsidrivers")
+	operatorClient, dynamicInformers, err := goc.NewClusterScopedOperatorClientWithConfigName(controllerConfig.KubeConfig, gvr, instanceName)
+	if err != nil {
+		return err
+	}
+
+	cm := manager.NewControllerManager()
```

Comment:
nit: Why not use the controllerset like the other CSI operators (and start the additional controller separately)?

Reply:
Reworked to `CSIControllerSet`. Initially, I needed extra arguments somewhere, but I don't remember where now.